tests/test_indexer.py

```diff
@@ -554,6 +556,66 @@ def test_multiple_schema_version_subscription_indexing_and_notification(self):
 
         self.delete_subscription(subscription_id)
 
+    def test_scrub_index_data(self):
+        with self.subTest("with schema version"):
+            'Extra fields are removed.'
+            manifest = read_bundle_manifest(self.blobstore, self.test_bucket, self.bundle_key)
+            index_data = create_index_data(self.blobstore, self.test_bucket, manifest)
+            index_data['files']['assay_json']['extra_top'] = 123
+            index_data['files']['assay_json']['extra_obj'] = {"something": "here", "another": 123}
+            index_data['files']['assay_json']['extra_lst'] = ["a", "b"]
+            index_data['files']['assay_json']['core']['extra_internal'] = 123
+            with self.assertLogs(logger, level="INFO"):
+                scrub_index_data(index_data['files'], self.bundle_key, logger)
+
+            self.verify_index_document_structure_and_content(
+                index_data,
+                self.bundle_key,
+                files=smartseq2_paried_ends_indexed_file_list,
+            )
+
+        with self.subTest("without schema version"):
+            'Only the manifest should exist.'
+            bundle_key = self.load_test_data_bundle_for_path(
+                "fixtures/indexing/bundles/unversioned/smartseq2/paired_ends_extras")
+            manifest = read_bundle_manifest(self.blobstore, self.test_bucket, bundle_key)
+            index_data = create_index_data(self.blobstore, self.test_bucket, manifest)
+            files = index_data['files'].keys()
+            for file in files:
+                index_data['files'][file].pop('core', None)
+            scrub_index_data(index_data['files'], bundle_key, logger)
+
+            self.assertEqual(3, len(index_data.keys()))
+            self.assertEqual("new", index_data['state'])
+            self.assertIsNotNone(index_data['manifest'])
+            self.assertIsNotNone(index_data['files'])
+            self.assertEqual(len(index_data['files']), 0)
+
+            expected_index_data = generate_expected_index_document(self.blobstore,
+                                                                   self.test_bucket,
+                                                                   bundle_key,
+                                                                   smartseq2_paried_ends_indexed_file_list)
+            if expected_index_data != index_data:
+                logger.error(f"Expected index document: {json.dumps(expected_index_data, indent=4)}")
+                logger.error(f"Actual index document: {json.dumps(index_data, indent=4)}")
+                self.assertDictEqual(expected_index_data, index_data)
```

Comment:
Nit: You know at this point that `assertDictEqual` **should** fail (and there must be a reason that you don't just use `self.fail("…")`) but I would add `assert False` below this.

Reply:
using assertDictEqual with a msg